src/scheduler/scheduler.rs

```diff
@@ -230,6 +205,29 @@ impl<VM: VMBinding> GCWorkScheduler<VM> {
                     .add(ForwardFinalization::<C::ProcessEdgesWorkType>::new());
             }
         }
+
+        // We add the VM-specific weak ref processing work regardless of MMTK-side options,
+        // including Options::no_finalizer and Options::no_reference_types.
+        //
+        // VMs need weak reference handling to function properly.  The VM may treat weak references
+        // as strong references, but it is not appropriate to simply disable weak reference
+        // handling from MMTk's side.  The VM, however, may choose to do nothing in
+        // `Collection::process_weak_refs` if appropriate.
+        //
+        // It is also not sound for MMTk core to turn off weak
+        // reference processing or finalization alone, because (1) not all VMs have the notion of
+        // weak references or finalizers, so it may not make sence, and (2) the VM may
+        // processing them together.
+
+        // VM-specific weak ref processing
```

Comment:
Maybe explain very briefly why `VMProcessWeakRefs` is a 'sentinel'.

Reply:
Done.

p.s. Strictly speaking, the first `VMProcessWeakRefs` packet added here doesn't have to be a sentinel, because it is the only packet in that bucket at that moment.  But I set it as sentinel anyway to avoid confusion, so that `VMProcessWeakRefs` is always a sentinel.